sdk/keyvault/azure-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateClient.java

```diff
@@ -0,0 +1,1259 @@
+     * @return The {@link DeletedCertificate deleted certificate}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public DeletedCertificate getDeletedCertificate(String name) {
+        return getDeletedCertificateWithResponse(name, Context.NONE).value();
+    }
+
+    /**
+     * Retrieves information about the specified deleted certificate. The GetDeletedCertificate operation  is applicable for soft-delete
+     * enabled vaults and additionally retrieves deleted certificate's attributes, such as retention interval, scheduled permanent deletion and the current deletion recovery level. This operation
+     * requires the certificates/get permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p> Gets the deleted certificate from the key vault enabled for soft-delete. Prints out the
+     * deleted certificate details when a response has been received.</p>
+     *
+     * {@codesnippet com.azure.security.keyvault.certificates.async.certificateclient.getDeletedCertificateWithResponse#string}
+     *
+     * @param name The name of the deleted certificate.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @throws ResourceNotFoundException when a certificate with {@code name} doesn't exist in the key vault.
+     * @throws HttpRequestException when a certificate with {@code name} is empty string.
+     * @return A {@link Response} whose {@link Response#value() value} contains the {@link DeletedCertificate deleted certificate}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<DeletedCertificate> getDeletedCertificateWithResponse(String name, Context context) {
+        return client.getDeletedCertificateWithResponse(name, context).block();
+    }
+
+    /**
+     * Permanently deletes the specified deleted certificate without possibility for recovery. The Purge Deleted Certificate operation is applicable for
+     * soft-delete enabled vaults and is not available if the recovery level does not specify 'Purgeable'. This operation requires the certificate/purge permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Purges the deleted certificate from the key vault enabled for soft-delete. Prints out the
+     * status code from the server response when a response has been received.</p>
+
+     * {@codesnippet com.azure.security.keyvault.certificates.async.certificateclient.purgeDeletedCertificate#string}
+     *
+     * @param name The name of the deleted certificate.
+     * @throws ResourceNotFoundException when a certificate with {@code name} doesn't exist in the key vault.
+     * @throws HttpRequestException when a certificate with {@code name} is empty string.
+     * @return A {@link VoidResponse}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public VoidResponse purgeDeletedCertificate(String name) {
```

Comment:
The return type for this should just be `void` as there is another overload which returns `VoidResponse` and takes the `Context` as additional param, right?
I don't see any guidelines around this - cc @JonathanGiles

Reply:
Agree with @srnagar.

Reply:
updated